### ISSUE TITLE
MockServerClient verify for the request body doesn't work

### ISSUE THREAD

**Issue by OleksiiZaitsevGL** [jamesdbloom/mockserver#601]
Version that I use: 5.5.1
An example that I tried:
`mockServer.verify(request()
                        .withMethod("POST")
                        .withPath("/requestUri")
                        .withBody("this part is not WORK!"),
                VerificationTimes.once());`
The test is green with any body (body doesn't match but the test is still green)

Also I tried change the body "an_example_body" for your test [NettyPortForwardingSecureProxyIntegrationTest](https://github.com/jamesdbloom/mockserver/blob/604e57941b7b544d0fabc4a7e654f0e6c946ce9f/mockserver-netty/src/test/java/org/mockserver/integration/proxy/direct/NettyPortForwardingSecureProxyIntegrationTest.java) and still the test is green even if body not matched.

**Comment by jamesdbloom** [comment]
What test framework are you using?  Does if fail when an `java.lang.AssertionError` is thrown?  Can you please test as follows `throw new AssertionError("test exception")`.

**Comment by davidwcarlson** [comment]
I am also seeing this issue running MockServer 5.5.4 via JavaAPI (i.e., `startClientAndServer()`)

Here is some sample logging from the test.  Notice that the request has no body, while the expectation does, yet they match.  I have tried setting JSON-Strict, JSON-Only matching, and text for the body.

Expected is they do not match.

`2019-06-21 10:12:27 - request:

	{
	  "method" : "POST",
	  "path" : "/job-testJobId/testRealm-j-testJobId/v1/jobs/3/savepoints",
	  "headers" : {
	    "Accept" : [ "application/json" ],
	    "User-Agent" : [ "Swagger-Codegen/1.0.0/java" ],
	    "Content-Type" : [ "application/json" ],
	    "Host" : [ "localhost:53623" ],
	    "Connection" : [ "keep-alive" ],
	    "content-length" : [ "0" ]
	  },
	  "keepAlive" : true,
	  "secure" : false
	}

 matched expectation:

	{
	  "httpRequest" : {
	    "method" : "POST",
	    "path" : "/job-testJobId/testRealm-j-testJobId/v1/jobs/3/savepoints",
	    "body" : "{\"cancel-job\":true}"
	  },
	  "times" : {
	    "unlimited" : true
	  },
	  "timeToLive" : {
	    "unlimited" : true
	  },
	  "httpResponse" : {
	    "statusCode" : 500
	  }
	}
`

**Comment by davidwcarlson** [comment]
I also have verifications setup which pass, even though the request should not match.

**Comment by csbiggar** [comment]
I think I'm seeing the same thing, with 5.6.0 (this is my first time using mock-server)

I was trying out using mock-server to TDD creating a request to an external service. This worked fine for verify `withPath` - the test failed, then I changed the production code to make it pass. However, when I added the `.withBody(..)` verification which I expected to fail , the test continued to pass?

I'm using  Kotlin 1.3.41 (with http library Fuel 2.1.0) and Junit 5.3.1 

Here's a bare bones example project including my dependencies, test and implementation  - try running
https://github.com/csbiggar/mock-server-example

Thanks!
